src/devices/acurite.c

```diff
@@ -713,36 +714,95 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+
             // Battery status is the 7th bit 0x40. 1 = normal, 0 = low
             battery_low = (bb[2] & 0x40) == 0;
 
-            /* clang-format off */
-            data = data_make(
-                    "model",                "",             DATA_STRING, "Acurite-Tower",
-                    "id",                   "",             DATA_INT,    sensor_id,
-                    "channel",              NULL,           DATA_STRING, channel_str,
-                    "battery_ok",           "Battery",      DATA_INT,    !battery_low,
-                    "temperature_C",        "Temperature",  DATA_FORMAT, "%.1f C", DATA_DOUBLE, tempc,
-                    "humidity",             "Humidity",     DATA_FORMAT, "%u %%", DATA_INT,    humidity,
-                    "mic",                  "Integrity",    DATA_STRING, "CHECKSUM",
-                    NULL);
-            /* clang-format on */
+            if (message_type == ACURITE_MSGTYPE_LEAK_DETECTOR) {
+                int is_wet = (bb[3] & 0x10) >> 4;
+
+                /* clang-format off */
+                data = data_make(
+                        "model",                "",             DATA_STRING, "Acurite-Leak",
+                        "id",                   "",             DATA_INT,    sensor_id,
+                        "channel",              NULL,           DATA_STRING, channel_str,
+                        "battery_ok",           "Battery",      DATA_INT,    !battery_low,
+                        "leak_detected",        "Leak",         DATA_INT,    is_wet,
+                        "mic",                  "Integrity",    DATA_STRING, "CHECKSUM",
+                        NULL);
+                /* clang-format on */
+            }
+            else if (message_type == ACURITE_MSGTYPE_TOWER_SENSOR) {
+
+                // Humidity is stored in byte 3
+                // The value is directly encoded as %rH
+                // The possible values here are 0-128, but real values are only 1-99 %rH
+                // pIII IIII
+                humidity = (bb[3] & 0x7f);
+                if (humidity < 1 || humidity > 99) {
+                    if (decoder->verbose) {
+                        fprintf(stderr, "%s: Acurite TXR sensor 0x%04X Ch %s : Impossible humidity: %d %%rH\n",
+                                __func__, sensor_id, channel_str, humidity);
+                    }
+                    continue;
+                }
+
+                // temperature encoding used by "tower" sensors 592txr
+                // 14 bits available after removing both parity bits.
+                // 11 bits needed for specified range -40 C to 70 C (-40 F - 158 F)
+                // Possible ranges are -100 C to 1538.4 C, but most of that range
+                // is not possible on Earth.
+                int temp_raw = ((bb[4] & 0x7F) << 7) | (bb[5] & 0x7F);
+                tempc        = temp_raw * 0.1 - 100;
+                if (tempc < -40 || tempc > 70) {
```

Comment:
as long as this allows messages with -40 through, I think this should be ok.

Reply:
-40 should still come through as is here.

Reply:
No change here. -40 should still pass through, but -41 won't.